hack_parallel.0.1.1 fails to build on OpenBSD 6.6

There are two errors. 

(1) The C source is not including `<errno.h>` in `src/heap/hh_shared.c` which is required on BSD. 

The fix is to add something like this on line 107:

     #ifdef __OpenBSD__
     #include <errno.h>
     #endif

Including this header is likely required on FreeBSD and NetBSD as well, but I don't have a way to test those. 

(2) The C source `src/utils/sysinfo.c` is attempting to include a header which doesn't exist.

The fix is to change lines 16-20 to:

    #ifndef _WIN32
    #ifndef __APPLE__
    #ifndef __OpenBSD__
    #include <sys/sysinfo.h>
    #endif
    #endif 
    #endif

I'd submit a pull request, but I have no idea where the module sources live :/ sorry, C++ guy, not familiar with OCaml. 

## opam config report

    $ opam config report
    # opam config report
    # opam-version      2.0.5
    # self-upgrade      no
    # system            arch=x86_64 os=openbsd os-distribution=openbsd os-version=6.6
    # solver            builtin-mccs+glpk
    # install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed   
    # upgrade-criteria  -removed,-count[version-lag,solution],-new
    # jobs              1
    # repositories      1 (http) (default repo at ef627cdd)
    # pinned            4 (git)
    # current-switch    ocaml-system.4.09.0

## opam tool output

In https://github.com/comby-tools/comby directory, run `opam install --deps-only -y`

    #=== ERROR while compiling hack_parallel.0.1.1 ================================#
    # context     2.0.5 | openbsd/x86_64 | ocaml-system.4.09.0 | https://opam.ocaml.org#ef627cdd
    # path        ~/.opam/ocaml-system.4.09.0/.opam-switch/build/hack_parallel.0.1.1
    # command     /usr/local/bin/gmake libhp.a
    # exit-code   2
    # env-file    ~/.opam/log/hack_parallel-8994-ad6566.env
    # output-file ~/.opam/log/hack_parallel-8994-ad6566.out
    ### output ###
    # hh_shared.c:1755:53: error: use of undeclared identifier 'errno'
    # [...]
    # hh_shared.c:1763:53: error: use of undeclared identifier 'errno'
    #     printf("Error while collecting: %s\n", strerror(errno));
    #                                                     ^
    # hh_shared.c:1871:17: warning: assigning to 'value *' (aka 'long *') from 'const value *' (aka 'const long *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
    #   if (!exn) exn = caml_named_value("heap_full");
    #                 ^ 
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    # hh_shared.c:2010:17: warning: assigning to 'value *' (aka 'long *') from 'const value *' (aka 'const long *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
    #   if (!exn) exn = caml_named_value("hash_table_full");
    #                 ^ 
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
    # 8 warnings and 3 errors generated.
    # gmake: *** [Makefile:145: _build/src/heap/hh_shared.o] Error 2

--- a/packages/hack_parallel/hack_parallel.0.1.1/opam
+++ b/packages/hack_parallel/hack_parallel.0.1.1/opam
@@ -25,3 +25,4 @@ url {
   src: "https://github.com/rvantonder/hack-parallel/archive/v0.1.1.tar.gz"
   checksum: "md5=64485d8a2bad2751cb31255bc1be33b7"
 }
+available: os != "openbsd"

